tests/testthat/test-process_request.R

```diff
@@ -0,0 +1,86 @@
+test_that("process_request properly errors on wrong content type", {
```

Comment:
You might want to split these into a smaller number of tests. Why? so that your search is narrower when/if things fail.